types/coin.go

```diff
@@ -336,15 +336,25 @@ func (coins Coins) IsAllGT(coinsB Coins) bool {
 	return true
 }
 
-// IsAllGTE returns true iff for every denom in coins, the denom is present at
-// an equal or greater amount in coinsB.
+// IsAllGTE returns false if for any denom in coinsB,
+// the denom is present at a smaller amount in coins;
+// else returns true.
 func (coins Coins) IsAllGTE(coinsB Coins) bool {
-	diff, _ := coins.SafeSub(coinsB)
-	if len(diff) == 0 {
+	if len(coinsB) == 0 {
 		return true
 	}
 
-	return !diff.IsAnyNegative()
+	if len(coins) == 0 {
+		return false
+	}
+
+	for _, coinB := range coinsB {
```

Comment:
I think this is inefficient - if the coins have the same denoms, they should be in the same array order

Reply:
I'll open an issue re: making the comparison operators more efficient. Is that OK with you @cwgoes?

Reply:
That's fine, thanks.